source/reference/menus.rst

```diff
@@ -0,0 +1,105 @@
+===================
+ Menus - Reference
+===================
+
+.. seealso::
+
+   :doc:`Documentation on menus </customization/menus>`
+      Explains how menus work and what you can do
+
+
+File Format
+===========
+
+=============  ===========================================
+**Format**     JSON (with comments)
+
+**Extension**  ``.sublime-menu``
+
+**Name**       One out of the list of available menus.
+               See :ref:`menu-types` for the complete list
+               and what menu they represent.
+
+**Location**   Any
+
+**Content**    A list of :ref:`menu-item-objects`.
+=============  ===========================================
+
+
+Example
+*******
+
+The following is an excerpt
+from the default :file:`Main.sublime-menu` file.
+
+.. code-block:: json
+
+   [
+       {
+           "caption": "Edit",
+           "mnemonic": "E",
+           "id": "edit",
+           "children":
+           [
+               { "command": "undo", "mnemonic": "U" },
+               { "command": "redo_or_repeat", "mnemonic": "R" },
+               {
+                   "caption": "Undo Selection",
+                   "children":
+                   [
+                       { "command": "soft_undo" },
+                       { "command": "soft_redo" }
+                   ]
+               },
+               { "caption": "-", "id": "clipboard" },
+               { "command": "copy", "mnemonic": "C" },
+               { "command": "cut", "mnemonic": "t" },
+               { "command": "paste", "mnemonic": "P" },
+               { "command": "paste_and_indent", "mnemonic": "I" },
+               { "command": "paste_from_history", "caption": "Paste from History" }
+           ]
+       }
+   ]
+
```

Comment:
Not sure whether we need an extra empty line here. Not sure what we've been doing in these cases.